Redirect to original URL after inspecting

davea opened the pull request:
Fixes mysociety/fixmystreetforcouncils#204

dracos commented on perllib/FixMyStreet/App/Controller/Report.pm:
```diff
@@ -449,20 +453,23 @@ sub inspect : Private {
-                $params->{filter_category} = $categories if $categories;
-                $params->{js} = 1 if $c->get_param('js');
-                $redirect_uri = $c->uri_for( "/around", $params );
+                unless ($redirect_uri = $c->get_param("post_inspect_url")) {
```
Looks like this not being present never happens in test now.

dracos commented on perllib/FixMyStreet/App/Controller/Report.pm:
```diff
@@ -449,20 +453,23 @@ sub inspect : Private {
+            # We go here rather than the shortlist because it makes it much
+            # simpler to inspect many reports in the same location. The
+            # shortlist is always a single click away, being on the main nav.
             if ($c->user->has_body_permission_to('planned_reports')) {
```
As this overrides the work done before, should it perhaps come first in the previous if statement (I say first so that it doesn't have to bother doing the `owns_problem` check if this one passes).

davea commented:
Ta for the review. Good catch on the opportunity to remove a DB query! I've fixed that up and added a test for when `post_inspect_url` is empty.

Away until Tuesday so please do merge if you're happy with this!
